test/jdk/java/awt/print/Dialog/DialogType.java

```diff
@@ -26,49 +26,58 @@
+import javax.print.attribute.standard.DialogTypeSelection;
+
+import jtreg.SkippedException;
 
-  static String[] instructions = {
-     "This test assumes and requires that you have a printer installed",
-     "It verifies that the dialogs behave properly when using new API",
-     "to optionally select a native dialog where one is present.",
-     "Two dialogs are shown in succession.",
-     "The test passes as long as no exceptions are thrown, *AND*",
-     "if running on Windows only, the first dialog is a native windows",
-     "control which differs in appearance from the second dialog",
-     ""
-  };
+public class DialogType {
+    private static PrinterJob job;
 
-  public static void main(String[] args) {
+    private static final String INSTRUCTIONS = """
+        This test assumes and requires that you have a printer installed.
+        It verifies that the dialogs behave properly when using new API
+        to optionally select a native dialog where one is present.
+        Two dialogs are shown in succession.
+        The test passes as long as no exceptions are thrown.
+        Note: On Mac OS & on Windows, the first dialog is a native
+        control which differs in appearance from the second dialog
+    """;
 
-      for (int i=0;i<instructions.length;i++) {
-         System.out.println(instructions[i]);
-      }
+    public static void main(String[] args) throws Exception {
+        job = PrinterJob.getPrinterJob();
+        if (job.getPrintService() == null) {
+            throw new SkippedException("Test skipped, printer is unavailable");
+        }
+        PassFailJFrame passFailJFrame = PassFailJFrame.builder()
+            .instructions(INSTRUCTIONS)
+            .columns(40)
+            .build();
+        testDialogType();
+        passFailJFrame.awaitAndCheck();
+    }
 
-      PrinterJob job = PrinterJob.getPrinterJob();
-      if (job.getPrintService() == null) {
-         return;
-      }
+    private static void testDialogType() {
+        setPrintDialogAttributes(DialogTypeSelection.NATIVE);
+        setPrintDialogAttributes(DialogTypeSelection.COMMON);
+    }
 
-      PrintRequestAttributeSet aset = new HashPrintRequestAttributeSet();
-      aset.add(DialogTypeSelection.NATIVE);
-      job.printDialog(aset);
-      Attribute[] attrs = aset.toArray();
-      for (int i=0;i<attrs.length;i++) {
-          System.out.println(attrs[i]);
-      }
-      aset.add(DialogTypeSelection.COMMON);
-      job.printDialog(aset);
-      attrs = aset.toArray();
-      for (int i=0;i<attrs.length;i++) {
-          System.out.println(attrs[i]);
-      }
-   }
+    private static void setPrintDialogAttributes(DialogTypeSelection selection) {
+        PrintRequestAttributeSet aset = new HashPrintRequestAttributeSet();
+        aset.add(selection);
+        job.printDialog(aset);
+        Attribute[] attrs = aset.toArray();
+        for (int i = 0; i < attrs.length; i++) {
+            System.out.println(attrs[i]);
```

Comment:
Is it better to log the attributes using PFJ log area ?
or is it really require to log the output ?

Reply:
I believe this logging is done for the purpose of debugging and it not useful for the person running the test.

Reply:
Yes, that is correct. The log area is useful when there is a verification step - such as 'live events' tracking and validation that needs to be displayed on the UI and the test is marked as Pass or Fail as per that. 
Here, if the test fails for any reason - identifying that will be in the scope of the debugging the test by looking closely into the JTR - and hence we have the SOP to log it into the jtr file.